gossip/src/crds_value.rs

```diff
@@ -165,6 +165,52 @@ impl CrdsData {
     }
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, AbiExample)]
+pub struct AccountsHashes {
+    pub from: Pubkey,
+    pub hashes: Vec<(Slot, Hash)>,
+    pub wallclock: u64,
+}
+
+impl Sanitize for AccountsHashes {
+    fn sanitize(&self) -> Result<(), SanitizeError> {
+        sanitize_wallclock(self.wallclock)?;
+        for (slot, _) in &self.hashes {
+            if *slot >= MAX_SLOT {
+                return Err(SanitizeError::ValueOutOfBounds);
+            }
+        }
+        self.from.sanitize()
+    }
+}
+
+impl AccountsHashes {
+    pub fn new(from: Pubkey, hashes: Vec<(Slot, Hash)>) -> Self {
+        Self {
+            from,
+            hashes,
+            wallclock: timestamp(),
+        }
+    }
+
+    /// New random AccountsHashes for tests and benchmarks.
+    pub(crate) fn new_rand<R: Rng>(rng: &mut R, pubkey: Option<Pubkey>) -> Self {
+        let num_hashes = rng.gen_range(0, MAX_SNAPSHOT_HASHES) + 1;
+        let hashes = std::iter::repeat_with(|| {
+            let slot = 47825632 + rng.gen_range(0, 512);
+            let hash = solana_sdk::hash::new_rand(rng);
+            (slot, hash)
+        })
+        .take(num_hashes)
+        .collect();
+        Self {
+            from: pubkey.unwrap_or_else(pubkey::new_rand),
+            hashes,
+            wallclock: new_rand_timestamp(rng),
+        }
+    }
+}
+
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, AbiExample)]
 pub struct SnapshotHashes {
```

Comment:
If the new struct is identical to the old one, why not just rename it to `AccountHashes`?
which would also open room for a new `SnapshotHashes` struct.

So I am thinking something like:
* Rename `SnapshotHashes` struct to `AccountHashes` and use the new name for all account-hashes api.
* Also add the alias `type LegacySnapshotHashes = AccountHashes;` and use `LegacySnapshotHashes` for all current snapshot-hashes api.
* Future PR, add the new `SnapshotHashes` struct + the new enum variants etc.